src/main/java/io/reactivex/internal/schedulers/ScheduledRunnable.java

```diff
@@ -26,7 +26,12 @@
     private static final long serialVersionUID = -6120223772001106981L;
     final Runnable actual;
 
-    static final Object DISPOSED = new Object();
+    /** Indicates that the parent tracking this task has been notified about its completion. */
+    static final Object PARENT_DISPOSED = new Object();
+    /** Indicates the dispose() was called from within the run/call method. */
+    static final Object SYNC_DISPOSED = new Object();
+    /** Indicates the dispose() was called from another thread. */
+    static final Object ASYNC_DISPOSED = new Object();
```

Comment:
supernit: `ASYNC` doesn't mean another thread :)

But for name conciseness I guess it's fine